Ar_MethodHw.js

```diff
@@ -0,0 +1,185 @@
+//    let result = [];
+//    this.myForEach(element => {
+//
+//      result.push(anyT(element));
+//    })
+//
+//    return result;
+// }
+//
+//
+// let double = [1, 2, 3].myMap((el) => {
+//  return el * 2;
+// })
+// console.log(double);
+// 3.
+// Array.prototype.myFilter = function(callback){
+//     let result = [];
+//     this.myForEach(element => {
+//
+//       result.push(callback(element));
+//     })
+//
+//     return result;
+//   }
+//
+// let ages = [32, 33, 16, 40].myFilter((el)=>{
+//   return el >= 18;
+// })
+// console.log(ages)
+//
+// let ages = [32, 33, 16, 40].myFilter((el)=>{
+//     if ( el >= 18){
+//       console.log(el)
+//     }else{
+//      (el <= 18)
+//     }
+//     console.log()
+//   return el;
+// })
+// console.log(ages)
+
+// 4. The every() method checks if all elements in an array pass a test (provided as a function).
+//
+// The every() method executes the function once for each element present in the array:
+//
+// If it finds an array element where the function returns a false value, every() returns false (and does not check the remaining values)
+// If no false occur, every() returns true
+// Note: every() does not execute the function for array elements without values.
+//
+// Note: every() does not change the original array
+//4
+Array.prototype.myEvery = function(callback) {
+let result = [];
+ this.myForEach(element => {
+
+   result.push(callback(element));
+ })
+   if (!result) {
+     return false;
+   }
+ }
+ return true;
+}
+console.log([4,6,10,20,2].myEvery(element)
+
+// 5. Reduce (acc,currentValue)
+//MY ATTEMPT
+Array.prototype.myReduce = function (acc,currentValue = this[0]){
+  let acc = // element
+
+  return acc;
+}
+let array = [2,3,4,6];
+console.log(array.myReduce((a,b)=> {return a + b}))
+
+//LEARNING IN CLASS
+// Array.prototype.myReduce = function (callback,acc){
+//   let startingIdx = 0;
+//   if(!acc){
+//     acc = this[0];
+//     startingIdx ++;
+//   }
+//   for(let i=0;startingIdx;i++){
+//     // if we know that acc is starting at 0, we know that i should not start at 0.
+//     acc = callback(acc,this[i]);
+//   };
+//   return acc;
+// };
+//
+// let arr = [1,2,5,3,4];
+// let result = arr.myReduce((acc,currentEl)=> {
+//   return acc + currentEl;
+// })
+
+
+//.6
+
+//function transpose (mtx){
+//   let grid =[];
+//   for (let i=0;i<mtx[0].length;i++){ //mtx[0] grabbing the 1st elements length which is an array
+//         grid[i]=[];
+//     for (let j= 0 ;j<mtx.length;j++){
+//         grid[i][j]= mtx[j][i];// switching those elements
+//     }
+//   }
+//   return grid;
+// }
+//
+// let mtx = [[1,2,3],[1,2,3],[1,2,3]]
+// console.log(transpose(mtx));
+
+//.7
+// let arr = [1,2,3,4,5];
+// let newArr = arr.myMap(el => {return el + 10;})
+// console.log(newArr)
+//
+// //.8 The toString() method converts a number to a string.
+// let array = [2,4,6,7,10]
+// let newArray = array.myMap(el =>[return element,toString()]);
```

Comment:
Remember, we don't just want to convert it to a string, we want to convert it to an *empty* string